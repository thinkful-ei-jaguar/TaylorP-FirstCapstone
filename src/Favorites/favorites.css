.favorites-main {
  text-align: center;
  padding-top: 5px;
}