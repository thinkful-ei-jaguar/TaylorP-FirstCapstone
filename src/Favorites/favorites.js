import React, { Component } from "react";
import RecipeCard from "../Recipes/recipeCard";
import RecipeContext from "../Context/recipeContext";
import TokenService from "../Services/token-service";
import RecipeApiService from "../Services/recipe-api-service";
import "./favorites.css";

class Favorites extends Component {
  static contextType = RecipeContext;

  componentDidMount() {
    const id = TokenService.getUserId();
    this.context.clearError();
    RecipeApiService.getFavorites(id)
      .then(this.context.setFavoriteList)
      .catch(this.context.setError);
  }

  setSpiritType = (e) => {
    const filter = e.target.value;
    this.context.setSpirit(filter);
  };

  renderFavorites() {
    const { favoriteList = [] } = this.context;
    const filter = this.context.spirit;
    let filteredList = [];

    if (filter === "all") {
      filteredList = favoriteList;
    } else {
      filteredList = favoriteList.filter(
        (spirit) => spirit.spirit_id === Number(filter)
      );
    }

    return filteredList.map((recipe) => (
      <li key={recipe.favorite_id}>
        <RecipeCard recipe={recipe} />
      </li>
    ));
  }

  render() {
    return (
      <main className="favorites-main">
        <h2>Your Favorite Cocktails</h2>
        <div className="fav-search">
          <label htmlFor="spirit-search">Spirit:</label>
          <select
            name="spirit-search"
            id="spirit-search"
            onChange={this.setSpiritType}
          >
            <option value="all">All</option>
            <option value="2">Vodka</option>
            <option value="3">Gin</option>
            <option value="1">Tequila</option>
            <option value="4">Bourbon</option>
            <option value="7">Rye</option>
            <option value="5">Scotch</option>
            <option value="6">Amaro</option>
            <option value="8">Cognac</option>
            <option value="9">Rum</option>
          </select>
        </div>
        <ul className="favorites-list" aria-live="polite">
          {this.renderFavorites()}
        </ul>
      </main>
    );
  }
}

export default Favorites;
