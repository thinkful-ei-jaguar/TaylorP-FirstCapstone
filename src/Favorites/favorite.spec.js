import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'
import toJson from 'enzyme-to-json'
import Favorites from './favorites'
import RecipeContext from '../Context/recipeContext'

describe('<Favorites />', () => {
  it('renders without crashing', () => {
    const match = {params: {id: 1}}
    mount(
      <MemoryRouter>
        <Favorites match={match}/>
      </MemoryRouter>
    )
  })
  it('renders as expected', () => {
    const match = { params: { id: 1 } }
    const tree = renderer.create(
      <MemoryRouter>
        <Favorites match={match}/>
      </MemoryRouter>
    ).toJSON()
    expect(tree).toMatchSnapshot();
  })
  it('renders as expected with favorites', () => {
    const match = { params: { id: 1 } }
    const ctxval = {
      recipe: {
        id: 1,
        recipe_name: 'test-recipe',
        recipe_ingredients: ['step1','step2','step3'],
        recipe_prep: 'do the thing and test it',
        recipe_img: null,
      },
      favoriteList: [
        {
          favorite_id: 1,
          id:2,
          recipe_name: 'test fave 1',
          spirit_id: 2,
          user_id: 1
        },
        {
          favorite_id: 2,
          id:3,
          recipe_name: 'test fave 2',
          spirit_id: 3,
          user_id: 1
        }
      ],
      spirit: 'all',
      error: null,
      setError: () => {},
      clearError: ()=> {},
      setRecipe: () => {},
      clearRecipe: ()=> {},
      setFavoriteList: () => {},
      setSpirit: () => {},
      addFavorite: () => {}
    }
    const tree = renderer.create(
      <MemoryRouter>
        <RecipeContext.Provider value={ctxval}>
          <Favorites match={match}/>
        </RecipeContext.Provider>
      </MemoryRouter>
    ).toJSON()
    expect(tree).toMatchSnapshot();
  })
})