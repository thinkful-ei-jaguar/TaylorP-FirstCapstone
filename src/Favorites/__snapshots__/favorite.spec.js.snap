// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Favorites /> renders as expected 1`] = `
<main
  className="favorites-main"
>
  <h2>
    Your Favorite Cocktails
  </h2>
  <label
    htmlFor="spirit-search"
  >
    Spirit:
  </label>
  <select
    id="spirit-search"
    name="spirit-search"
    onChange={[Function]}
  >
    <option
      value="all"
    >
      All
    </option>
    <option
      value="2"
    >
      Vodka
    </option>
    <option
      value="3"
    >
      Gin
    </option>
    <option
      value="1"
    >
      Tequila
    </option>
    <option
      value="4"
    >
      Bourbon
    </option>
    <option
      value="7"
    >
      Rye
    </option>
    <option
      value="5"
    >
      Scotch
    </option>
    <option
      value="6"
    >
      Amaro
    </option>
    <option
      value="8"
    >
      Cognac
    </option>
    <option
      value="9"
    >
      Rum
    </option>
  </select>
  <ul
    aria-live="polite"
    className="favorites-list"
  />
</main>
`;

exports[`<Favorites /> renders as expected with favorites 1`] = `
<main
  className="favorites-main"
>
  <h2>
    Your Favorite Cocktails
  </h2>
  <label
    htmlFor="spirit-search"
  >
    Spirit:
  </label>
  <select
    id="spirit-search"
    name="spirit-search"
    onChange={[Function]}
  >
    <option
      value="all"
    >
      All
    </option>
    <option
      value="2"
    >
      Vodka
    </option>
    <option
      value="3"
    >
      Gin
    </option>
    <option
      value="1"
    >
      Tequila
    </option>
    <option
      value="4"
    >
      Bourbon
    </option>
    <option
      value="7"
    >
      Rye
    </option>
    <option
      value="5"
    >
      Scotch
    </option>
    <option
      value="6"
    >
      Amaro
    </option>
    <option
      value="8"
    >
      Cognac
    </option>
    <option
      value="9"
    >
      Rum
    </option>
  </select>
  <ul
    aria-live="polite"
    className="favorites-list"
  >
    <a
      href="/recipes/2"
      onClick={[Function]}
    >
      <li
        className="recipe-li"
      >
        <h2>
          test fave 1
        </h2>
      </li>
    </a>
    <a
      href="/recipes/3"
      onClick={[Function]}
    >
      <li
        className="recipe-li"
      >
        <h2>
          test fave 2
        </h2>
      </li>
    </a>
  </ul>
</main>
`;
